File: LAB_4/pracownik.cpp
#include <iostream>
#include <string>

using namespace std;

class Pracownik {
protected:
    string stanowisko;
    float wynagrodzenie;

public:
    Pracownik(string stanowisko, float wynagrodzenie) : stanowisko(stanowisko), wynagrodzenie(wynagrodzenie) {}
};

class Nauczyciel : public Pracownik {
    string przedmiot;

public:
    Nauczyciel(string stanowisko, float wynagrodzenie, string przedmiot)
            : Pracownik(stanowisko, wynagrodzenie), przedmiot(przedmiot) {}

    void pokazDane() {
        cout << "Nauczyciel: \nStanowisko: " << stanowisko << "\nWynagrodzenie: " << wynagrodzenie << "\nPrzedmiot: " << przedmiot << endl;
    }
};

class Administracja : public Pracownik {
    string dzial;

public:
    Administracja(string stanowisko, float wynagrodzenie, string dzial)
            : Pracownik(stanowisko, wynagrodzenie), dzial(dzial) {}

    void pokazDane() {
        cout << "Pracownik Administracji: \nStanowisko: " << stanowisko << "\nWynagrodzenie: " << wynagrodzenie << "\nDział: " << dzial << endl;
    }
};

int main() {
    Nauczyciel nauczyciel("Nauczyciel glowny", 2100.0f, "Polski");
    Administracja administracja("Sekretarz", 8000.0f, "Dzial Obslugi");

    nauczyciel.pokazDane();
    administracja.pokazDane();

    return 0;
}

File: LAB_4/Pracownik.java
class Pracownik {
    protected String stanowisko;  // atrybut chroniony
    protected float wynagrodzenie;  // atrybut chroniony

    public Pracownik(String stanowisko, float wynagrodzenie) {
        this.stanowisko = stanowisko;
        this.wynagrodzenie = wynagrodzenie;
    }
}

class Nauczyciel extends Pracownik {
    private String przedmiot;

    public Nauczyciel(String stanowisko, float wynagrodzenie, String przedmiot) {
        super(stanowisko, wynagrodzenie);
        this.przedmiot = przedmiot;
    }

    public void pokazDane() {
        System.out.println("Stanowisko: " + stanowisko + ", Wynagrodzenie: " + wynagrodzenie + ", Przedmiot: " + przedmiot);
    }
}

class Administracja extends Pracownik {
    private String dzial;

    public Administracja(String stanowisko, float wynagrodzenie, String dzial) {
        super(stanowisko, wynagrodzenie);
        this.dzial = dzial;
    }

    public void pokazDane() {
        System.out.println("Stanowisko: " + stanowisko + ", Wynagrodzenie: " + wynagrodzenie + ", Dział: " + dzial);
    }
}

public class Main {
    public static void main(String[] args) {
        Nauczyciel nauczyciel = new Nauczyciel("Nauczyciel wspomagający", 5200.0f, "Matematyka");
        Administracja administracja = new Administracja("Sekretarz", 3200.0f, "Dział Obsługi");

        
        system.out.println("Nauczyciel:\n");
        nauczyciel.pokazDane();
        System.out.println("\nAdministracja:\n");
        administracja.pokazDane();
    }
}
